packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -1928,31 +1926,15 @@ function updateHostComponent(
     //
     // Once a fiber is upgraded to be stateful, it remains stateful for the
     // rest of its lifetime.
-    const newState = renderTransitionAwareHostComponentWithHooks(
+    renderTransitionAwareHostComponentWithHooks(
       current,
       workInProgress,
       renderLanes,
     );
-
-    // If the transition state changed, propagate the change to all the
-    // descendents. We use Context as an implementation detail for this.
-    //
-    // This is intentionally set here instead of pushHostContext because
-    // pushHostContext gets called before we process the state hook, to avoid
-    // a state mismatch in the event that something suspends.
```

Comment:
I'm not sure I fully understand this comment. If something suspends we pop back up. And the Component we're rendering (here: `TransitionAwareHostComponent`) should not be able to read the current Context just yet which is what pushing the context before the render would do, right? We can't nest Host Context anyway though. So this seems safe to move the push after the render.

Reply:
A common issue is that CompleteWork and UnwindWork unconditionally calls `popHostContext`. That's called even when something in the component suspends because it doesn't know if we suspended before or after the push. There's no state to indicate that.

That's why we always `pushHostContext` and other push things immediately when we enter the update before anything could error or suspend.

Since `renderTransitionAwareHostComponentWithHooks` might suspend, I don't think it's safe to move the `pushHostContext` after it. In fact I'm surprised we don't have a test that fails for that case. Maybe we're lacking coverage or there's some other thing that covers it but it shouldn't be safe in general.

Reply:
Makes sense. Moved the `pushContext` back to its original place. When the state changes, we update the context value which works because only a single transition provider is allowed.

Basically, we set the value on `pushHostContext` and when we update state.